### ISSUE TITLE
Guide on ADB commands

### ISSUE THREAD

**Issue by nesquena** [codepath/android_guides#217]
* <http://adbshell.com/commands>
* <https://developer.android.com/studio/command-line/shell.html>
* <https://developer.android.com/studio/command-line/adb.html>
* <http://www.androidcentral.com/android-201-10-basic-terminal-commands-you-should-know>
* <https://code.tutsplus.com/tutorials/android-adb-quick-guide--mobile-12456>
* <http://joyofandroid.com/what-is-adb-and-how-to-use-it/>
* <http://opensourceforgeeks.blogspot.com/2013/12/android-debug-bridge-adb.html>
* <https://www.youtube.com/watch?v=3wMlCucwGvE>
* <https://events.linuxfoundation.org/images/stories/pdf/lf_abs12_kobayashi.pdf>

**Comment by rogerhu** [comment]
https://docs.google.com/presentation/d/1VJ988Hdq-EAwsXBdLti-FtkxFPA0ZgXKvIAU90n9kUI/edit

**Comment by aimanbaharum** [comment]
If anyone could provide a one-line command for pulling data from a device, that would be the best :D

Normally, I would do as follows to pull a database file from app data:

```shell
$ adb shell
$ shell@user:/ $ run-as myapp.com.android
$ shell@user:/ $ cat databases/mydb.db > /mnt/sdcard/db.db
$ shell@user:/ $ exit
$ adb pull /mnt/sdcard/db.db .
```

**Comment by Hanif-abuvani** [comment]
If you are pulling your own app db then better to create the db on sdcard
only while creating the db in code, just give the path of sdcard, simple.

**Comment by aimanbaharum** [comment]
@Hanif-abuvani don't you think there will be a security issue with that approach? But it could be done by creating the db on debug mode only.

**Comment by Hanif-abuvani** [comment]
Yes, I am saying to go for this only at development time, while publishing
you can again change to conventional and more secure way. This is just a
way to skip a drudgery of pulling the db everytime via command.
